go/arrow/avro/common.go

```
@@ -0,0 +1,162 @@
+// handling. If stringsCanBeNull is true, then a string that matches one of the
+// nullValues set will be interpreted as NULL. Numeric columns will be checked
+// for nulls in all cases. If no nullValues arguments are passed in, the
+// defaults set in NewReader() will be kept.
+//
+// When no NULL values is given, the default set is taken from DefaultNullValues.
+func WithNullReader(stringsCanBeNull bool, nullValues ...string) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			cfg.stringsCanBeNull = stringsCanBeNull
+
+			if len(nullValues) == 0 {
+				nullValues = DefaultNullValues
+			}
+			cfg.nulls = make([]string, len(nullValues))
+			copy(cfg.nulls, nullValues)
+		default:
+			panic(fmt.Errorf("arrow/avro: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithColumnTypes allows specifying optional per-column types (disabling
+// type inference on those columns).
+//
+// Will panic if used in conjunction with an explicit schema.
+func WithColumnTypes(types map[string]arrow.DataType) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			if cfg.schema != nil {
+				panic(fmt.Errorf("%w: cannot use WithColumnTypes with explicit schema", arrow.ErrInvalid))
+			}
+			cfg.columnTypes = types
+		default:
+			panic(fmt.Errorf("%w: WithColumnTypes only allowed for csv reader", arrow.ErrInvalid))
+		}
+	}
+}
+
+// WithIncludeColumns indicates the names of the columns from the AVRO file
+// that should actually be read and converted (in the slice's order).
+// If set and non-empty, columns not in this slice will be ignored.
+//
+// Will panic if used in conjunction with an explicit schema.
+func WithIncludeColumns(cols []string) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			if cfg.schema != nil {
+				panic(fmt.Errorf("%w: cannot use WithIncludeColumns with explicit schema", arrow.ErrInvalid))
+			}
```

why do we need the explicit schema to do this? we have the avro schema so we can throw an error if the field is not found etc.

---

removed in latest commit